Bio/SeqIO/UniprotIO.py

```diff
@@ -495,8 +494,7 @@ def _parse_sequence(element):
                     self.ParsedSeqRecord.annotations["sequence_%s" % k] = int(v)
                 else:
                     self.ParsedSeqRecord.annotations["sequence_%s" % k] = v
-            seq = "".join(element.text.split())
-            self.ParsedSeqRecord.seq = Seq.Seq(seq, self.alphabet)
+            self.ParsedSeqRecord.seq = Seq("".join(element.text.split()))
             self.ParsedSeqRecord.annotations["molecule_type"] = "protein"
```

Comment:
Same here.

Reply:
No, these are records from UniProt - they are most definitely protein and not nucleotide sequences.

Reply:
That is true, but what is the purpose of storing the `molecule_type` here? As discussed; within Biopython the meaning of `molecule_type` is undefined.

Reply:
One reason for recording a consistent molecule type on parsing is to make conversion between file formats as easy as possible - as 10f1bfbaf2d976ade3509290097aea5968a3ea25 shows by dropping the DNA/RNA molecule type in parsing Ace, Gck, SnapGene (etc) we make it hard to convert to EMBL, GenBank, Nexus, SeqXML etc.

Reply:
It may work for UniProt, but this doesn't work in general, and is asking for trouble. Using one of our test files as an example, using the `molecule_type` set on parsing results in an invalid SeqXML file:
```python
>>> from Bio.Seq import Seq
>>> from Bio.SeqRecord import SeqRecord
>>> from Bio import SeqIO
>>> from io import BytesIO
>>> record = SeqIO.read('iro.gb', 'genbank')
>>> record.seq
Seq('CACAGGCCCAGAGCCACTCCTGCCTACAGGTTCTGAGGGCTCAGGGGACCTCCT...AAA')
>>> record.annotations['molecule_type']
'mRNA'
>>> handle = BytesIO()
>>> SeqIO.write([record], handle, 'seqxml')
>>> handle.seek(0)
>>> print(handle.read().decode())
<?xml version="1.0" encoding="utf-8"?>
<seqXML xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.seqxml.org/0.4/seqxml.xsd" seqXMLversion="0.4"><entry id="AL109817.1" source="human"><description>Homo sapiens mRNA full length insert cDNA clone EUROIMAGE 125195</description><RNAseq>CACAGGCCCAGAGCCACTCCTGCCTACAGGTTCTGAGGGCTCAGGGGACCTCCTGGGCCCTCAGGCTCTTTAGCTGAGAATAAGGGCCCTGAGGGAACTACCTGCTTCTCACATCCCCGGGTCTCTGACCATCTGCTGTGTGCCCCGACCCCCCCTACCCTGCTCCTCCACCA
```

Reply:
That's #3010 isn't it? DNA sequence (T not U) but annotation says RNA, known to happen in GenBank and EMBL files.